File: test1.go
package main;
import (
	"fmt"
	"time"
);

func pi(iterations int64) float64 {
	var dop int64 = 1;
	total := 0.0;
	for range iterations {
		total += 1.0 / float64(dop);
		// if you think those manipulations doesn't make any sense, arisnow thinks ohterwise (fuck off Griesson)
		dop += 2;
		total -= 1.0 / float64(dop);
		dop += 2;
	}
	return total * 4;
}

func main() {
	start := time.Now();
	fmt.Println(pi(100_000_000_000))
	endTime := time.Since(start);
	fmt.Println("time required: ", endTime);
} // By Griesson

File: test1.rs
use std::time::{SystemTime};

fn pi(iterations: i64) -> f64 {
    let mut dop: f64 = 1.;
    let mut result: f64 = 0.;
    for _ in 0..iterations {
        result += 1./dop;
        dop += 2.;
        result -= 1./dop;
        dop += 2.;
    }
    result * 4.
}

fn main() {
    let current = SystemTime::now();
    let pi = pi(100_000_000_000);
    let time_required = SystemTime::now().duration_since(current);
    println!("Pi: {pi}\nTime required: {time_required:?}");
} // By Griesson
